File: backend/src/main/java/com/backend/ord/seeders/factories/UserMockFactory.java
package com.backend.ord.seeders.factories;

import com.backend.ord.domain.entities.User;
import com.backend.ord.enums.UserRole;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class UserMockFactory extends AbstractFactory {
    private final PasswordEncoder passwordEncoder;

    public UserMockFactory(PasswordEncoder passwordEncoder) {
        this.passwordEncoder = passwordEncoder;
    }

    public User mockEntity() {
        return User.builder()
                .name(faker.name().fullName())
                .email(faker.internet().emailAddress())
                .password(passwordEncoder.encode(faker.internet().password()))
                .role(UserRole.USER)
                .build();
    }

    public User mockEntityWithCredentials(String email, String password) {
        return User.builder()
                .name(faker.name().fullName())
                .email(email)
                .password(passwordEncoder.encode(password))
                .role(UserRole.USER)
                .build();
    }
}

File: backend/src/main/kotlin/com/backend/ord/seeders/factories/UserMockFactory.kt
package com.backend.ord.seeders.factories

import com.backend.ord.core.user.model.UserEntity
import com.backend.ord.core.user.model.enums.UserRole
import com.backend.ord.seeders.EnumValuesGenerator
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class UserMockFactory(
    private val passwordEncoder: PasswordEncoder
) : AbstractFactory() {
    fun mockEntity(
        name: String = faker.name().fullName(),
        email: String = faker.internet().emailAddress(),
        password: String = faker.internet().password(),
        role: UserRole = UserRole.USER
    ): UserEntity {
        return UserEntity(
            name = name,
            email = email,
            password = passwordEncoder.encode(password),
            role = role,
            nativeLanguage = EnumValuesGenerator.mockLanguageName()
        )
    }
}
